castle/src/main/java/io/castle/android/Castle.java

```diff
@@ -158,71 +171,55 @@ public static void configure(Application application) {
     }
 
     /**
-     * Track event with a specified name
+     * Track custom event with a specified name
      * @param event Event name
      */
-    protected static void track(String event) {
+    public static void custom(String event) {
         if (event == null || event.isEmpty()) {
             return;
         }
-        track(new Event(event));
-    }
-
-    protected static void track(Event event) {
-        instance.eventQueue.add(event);
-    }
-
-    /**
-     * Track identify event with specified user identity. User identity will be persisted. A call to identify or reset will clear the stored user identity.
-     * @param userId user id
-     */
-    public static void identify(String userId) {
-        identify(userId, new HashMap<>());
+        track(new Custom(event));
     }
 
     /**
-     * Track identify event with specified user identity. User identity will be persisted. A call to identify or reset will clear the stored user identity. Provided user traits will be included in the identify event sent to the Castle API.
-     * @param userId user id
-     * @param traits user traits
+     * Track custom event with a specified name
+     * @param event Event name
      */
-    public static void identify(String userId, Map<String, String> traits) {
-        if (userId == null || userId.isEmpty() || traits == null) {
+    public static void custom(String event, Map<String, Object> properties) {
+        if (event == null || event.isEmpty()) {
             return;
         }
+        track(new Custom(event, properties));
+    }
 
-        // Log warning if identify is called without secure mode signature set.
-        if (!Castle.secureModeEnabled()) {
-            CastleLogger.w("Identify called without secure mode signature set. If secure mode is enabled in Castle and identify is called before secure, the identify event will be discarded.");
-        }
-
-        Castle.userId(userId);
-        track(new IdentifyEvent(userId, traits));
-        flush();
+    private static void track(Event event) {
+        instance.eventQueue.add(event);
     }
 
     /**
-     * Set user id
-     * @param userId  user id
+     * Set user information with specified jwt encoded user. User jwt will be persisted.
+     * @param userJwt encoded user jwt
      */
-    private static void userId(String userId) {
-        instance.storageHelper.setUserId(userId);
+    public static void userJwt(String userJwt) {
+        if (userJwt != null && !userJwt.isEmpty()) {
+            instance.storageHelper.setUserJwt(userJwt);
+        }
     }
 
     /**
      * Get user id from last identify call, returns null if not set
      * @return user id
      */
-    public static String userId() {
-        return instance.storageHelper.getUserId();
+    public static String userJwt() {
```

Comment:
Why do we need a getter for `userJwt`?

Reply:
We need it in the encoding but we can make it private

Reply:
It needs to be public because the class that needs it is not in the same package.